src/theory/bags/inference_generator.h

```diff
@@ -50,15 +50,14 @@ class InferenceGenerator
   /**
    * @param n is (bag x c) of type (Bag E)
    * @param e is a node of type E
-   * @return an inference that represents the following implication
-   * (=>
-   *   true
-   *   (= (bag.count e skolem) c))
-   *   if e is exactly node x. Node skolem is a fresh variable equals (bag x c).
-   *   Otherwise the following inference is returned
-   * (=>
-   *   true
-   *   (= (bag.count e skolem) (ite (= e x) c 0)))
+   * @return an inference that represents the following cases:
+   * 1- e, x are in the same equivalent class, then we infer:
+   *    (= (bag.count e skolem) (ite (>= c 1) c 0)))
+   * 2- e, x are known to be disequal, then we infer:
+   *    (= (bag.count e skolem) 0))
+   * 3- if neither holds, we infer:
+   *    (= (bag.count e skolem) (ite (and (= e x) (>= c 1)) c 0)))
```

Comment:
This reminds me of the splitting lemma for the join operator.

Reply:
I think this version is fine.

The challenge is that the theory of bags will not be informed of the direction that is split, since c >= 1 belongs to arithmetic.